### PR TITLE
hammer repository update gpg and publish method

### PR DESCRIPTION
**Pull request by pondrejk**
test results:
```
nosetests tests/foreman/cli/test_repository.py:RepositoryTestCase.test_positive_update_gpg_key -v 
Update the original gpg key ... ok

----------------------------------------------------------------------
Ran 1 test in 41.165s

OK

```
```
nosetests tests/foreman/cli/test_repository.py:RepositoryTestCase.test_positive_update_publish_method -v 
Update the original publishing method ... ok

----------------------------------------------------------------------
Ran 1 test in 26.420s

OK

```

### CODE REVIEW

File: tests/foreman/cli/test_repository.py [hunk 1]
```diff
@@ -837,28 +836,44 @@ def test_negative_update_auth_url_too_long(self):
...
         @id: 367ff375-4f52-4a8c-b974-8c1c54e3fdd3
 
         @Assert: Repository gpg key is updated
-
-        @caseautomation: notautomated
         """
+        gpg_key = make_gpg_key({'organization-id': self.org['id']})
+        gpg_key_new = make_gpg_key({'organization-id': self.org['id']})
+        new_repo = self._make_repository({
+            u'gpg-key-id': gpg_key[u'id'],
```
Comment (oshtaier) [inline tests/foreman/cli/test_repository.py]: We do not need to perform utf transformation for dict keys
`u'id'`->`'id'`

File: tests/foreman/cli/test_repository.py [hunk 2]
```diff
@@ -837,28 +836,44 @@ def test_negative_update_auth_url_too_long(self):
...
+        })
+        Repository.update({
+            u'id': new_repo[u'id'],
+            u'gpg-key-id': gpg_key_new[u'id'],
```
Comment (oshtaier) [inline tests/foreman/cli/test_repository.py]: Same here

File: tests/foreman/cli/test_repository.py [hunk 3]
```diff
@@ -837,28 +836,44 @@ def test_negative_update_auth_url_too_long(self):
...
 
         @Assert: Repository publishing method is updated
-
-        @caseautomation: notautomated
         """
+        new_repo = self._make_repository({
+            u'publish-via-http': u'no',
+        })
+        Repository.update({
+            u'id': new_repo[u'id'],
```
Comment (oshtaier) [inline tests/foreman/cli/test_repository.py]: same here

**Comment by pondrejk** [comment]
Changes requested by @oshtaier completed

**Pull request merged**